qemu/tests/hugepage_reset.py

```diff
@@ -0,0 +1,74 @@
+from avocado.utils import process
+
+from virttest import env_process
+from virttest import test_setup
+from virttest import utils_test
+from virttest import utils_misc
+from virttest.staging import utils_memory
+
+
+def run(test, params, env):
+    """
+    Re-assign nr-hugepages
+
+    1. Set up hugepage with 1G page size and hugepages=N
+    2. Boot up guest using /mnt/hugetlbfs/ as backend in QEMU CML
+    3. Change the nr_hugepages after the guest boot up (M < N and M > N)
+    4. Run the stress test **inside guest**
+    5. change the nr_hugepages after the stress test (M < N and M > N)
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def set_hugepage():
+        """Set nr_hugepages"""
+        process.run('echo 1 > /proc/sys/vm/compact_memory')
+        hp_config.target_hugepages = h_size
+        hp_config.set_hugepages()
+        if params.get('on_numa_node'):
+            hp_config.set_node_num_huge_pages(h_size, target_node,
+                                              hugepage_size)
```

Comment:
The setting process is covered by hp_config.setup().

Reply:
@huangyum Yes, but it does not use the `target_hugepages` but calculate a value it thinks proper, I am not sure if that's what we want? Or like here at Line 28 I set it to the fixed value we want.

Reply:
@huangyum Oh it does not, that's `__init__`, but `hp_config.setup()` has an extra step `mount_hugepage_fs`, do we need that every time setting hugepages?

Reply:
No, we don't have to. Mount works until unmount.

Reply:
@huangyum If we use `hp_config.setup()` it will mount it again and again, so keep this function instead of that?

Reply:
Sure.